src/storage/seg/src/item/mod.rs

```diff
@@ -83,6 +89,99 @@ impl std::fmt::Debug for Item {
     }
 }
 
+/// Items are the base unit of data stored within the cache.
+pub struct RichItem {
+    cas: u32,
+    age: u32,
+    item_info: u64,
+    item_info_ptr: *const u64,
+    raw: RawItem,
+}
+
+impl RichItem {
+    /// Creates a new `Item` from its parts
+    pub(crate) fn new(
+        raw: RawItem,
+        age: u32,
+        item_info: u64,
+        item_info_ptr: *const u64,
+        cas: u32,
+    ) -> Self {
+        RichItem {
+            cas,
+            age,
+            item_info,
+            item_info_ptr,
+            raw,
+        }
+    }
+
+    /// If the `magic` or `debug` features are enabled, this allows for checking
+    /// that the magic bytes at the start of an item match the expected value.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the magic bytes are incorrect, indicating that the data has
+    /// become corrupted or the item was loaded from the wrong offset.
+    pub(crate) fn check_magic(&self) {
+        self.raw.check_magic()
+    }
+
+    /// Borrow the item key
+    pub fn key(&self) -> &[u8] {
+        self.raw.key()
+    }
+
+    /// Borrow the item value
+    pub fn value(&self) -> Value {
+        self.raw.value()
+    }
+
+    /// CAS value for the item
+    pub fn cas(&self) -> u32 {
+        self.cas
+    }
+
+    pub fn age(&self) -> u32 {
+        self.age
+    }
+
+    // used to support multi readers and single writer
+    // return true, if the item is evicted/updated since being
+    // read from the hash table
+    pub fn is_not_changed(&self) -> bool {
+        unsafe { return self.item_info == *self.item_info_ptr & !FREQ_MASK }
+    }
+
+    /// Borrow the optional data
+    pub fn optional(&self) -> Option<&[u8]> {
+        self.raw.optional()
+    }
+
+    /// Perform a wrapping addition on the value. Returns an error if the item
+    /// is not a numeric type.
+    pub fn wrapping_add(&mut self, rhs: u64) -> Result<(), SegError> {
+        self.raw.wrapping_add(rhs)
+    }
+
+    /// Perform a saturating subtraction on the value. Returns an error if the
+    /// item is not a numeric type.
+    pub fn saturating_sub(&mut self, rhs: u64) -> Result<(), SegError> {
+        self.raw.saturating_sub(rhs)
+    }
+}
+
+impl std::fmt::Debug for RichItem {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::result::Result<(), std::fmt::Error> {
+        f.debug_struct("Item")
+            .field("cas", &self.cas())
+            .field("raw", &self.raw)
+            .field("item_info", &self.item_info)
+            .field("item_info_ptr", &self.item_info_ptr)
```

Comment:
Doesn't this print the ptr address? Is that really useful?

Reply:
yes, it is not useful, printing the value itself is not very useful as well. I can print all three (two value, one pointer) or we can remove the print. What do you think?

Reply:
It seems like printing if it's stale or not would maybe have value here?